specification/hugr.md

```diff
@@ -868,12 +869,39 @@ a valid type as long as the TypeArgs match the declared TypeParams, which can be
 (Note that within a polymorphic type scheme, type variables of kind `Sequence` or `Opaque` will only be usable
 as arguments to Opaque types---see [Extension System](#extension-system).)
 
+#### Row Variables
+
+Type variables of kind `TypeParam::List(TypeParam::Type(_))` are known as
+"row variables" and along with type parameters of the same kinds are given special
+treatment, as follows:
+* A `TypeParam` of such kind may be instantiated with not just a `TypeArg::List`
+  but also a single `TypeArg::Type`. (This is purely a notational convenience.)
+  For example, `Type::Function(usize, unit, <exts>)` is equivalent shorthand
+  for `Type::Function(#(usize), #(unit), <exts>)`.
+* When a `TypeArg::Sequence` is provided as argument for such a TypeParam, we allow
+  elements to be a mixture of both types (including variables of kind
+  `TypeParam::Type(_)`) and also row variables. When such variables are instantiated
+  (with other Sequences) the elements of the inner Sequence are spliced directly into
+  the outer (concatenating their elements), eliding the inner (Sequence) wrapper.
+
+For example, a polymorphic FuncDefn might declare a row variable X of kind
+`TypeParam::List(TypeParam::Type(Copyable))` and have as output a (tuple) type
+`Sum([#(X, usize)])`. A call that instantiates said type-parameter with
+`TypeArg::Sequence([usize, unit])` would then have output `Sum([#(usize, unit, usize)])`.
+
+See [Declarative Format](#declarative-format) for more examples.
+
+Note that since a row variable does not have kind Type, it cannot be used as the type of an edge.
+
 ### Extension Tracking
 
 The type of `Function` includes a set of [extensions](#extension-system) which are required to execute the graph.
-Every node in the HUGR is annotated with the set of extensions required to produce its inputs,
-and each operation provides (a way to compute) the set of extensions required to execute the node, known as the "delta";
-the union of these two must match the set of extensions on each successor node.
+Similarly, every dataflow node in the HUGR has a set of extensions required to execute the node (computed from its operation), also known as the "delta". The delta of any node must be a subset of its parent,
+except for FuncDefn's:
+* the delta of any child of a FuncDefn, must be a subset of the extensions in the FuncDefn's *type*
```

Comment:
This is a bit awkward. Suggestions welcome

Reply:
just remove the comma I think